src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -361,29 +328,31 @@
     (let [population-registry (get-in metrics/metrics-registries [:population :registry])]
       (pop/initialize-population-metrics! population-registry read-db))
 
-    (when (.exists discard-dir)
-      (dlo/create-metrics-for-dlo! discard-dir))
-    (let [broker (try
-                   (log/info "Starting broker")
-                   (mq/build-and-start-broker! "localhost"
-                                               (conf/mq-dir config)
-                                               command-processing)
-                   (catch java.io.EOFException e
-                     (log/error
-                      "EOF Exception caught during broker start, this "
-                      "might be due to KahaDB corruption. Consult the "
-                      "PuppetDB troubleshooting guide.")
-                     (throw e)))
+    ;; Error handling here?
+    (let [stockdir (conf/stockpile-dir config)
+          command-chan (async/chan
+                         (if (:catalog-facts-bash command-processing)
```

Comment:
And we only mark the commands as deleted so that we can avoid doing anything involving the messy filesystem world while holding the queue locks.  The deletion is handled (more easily) by the thread that eventually pulls the deleted command off the queue.